### PR TITLE
Pomo progress bar

### PR DESCRIPTION
**Pull request by qho-queenie**
The goals of this PR:
1. Added a circular progress bar for each session. 
There are a lot of changes in this PR, cause of linting and files changing from .js to .jsx, but most of this new logic is in `<CircularProgressBar />`, a little bit in `<ClockFace />`, and `CircularProgressBar.scss`.
Screen Shot 2021-11-08 at 6 43 32 PM

2. Modal is no longer a portal but a regular child component. The linter keeps throwing errors saying it is a circular dependency, so I loosely coupled the modal as `<ClockFace />`'s child in App.jsx, and I don't think portal can do this.

3. Added prototypes checking in each of the components.

4. Added eslint, stylelint (recommended and airbnb extensions), react-hooks, jsx-a11y, etc. Husky and lint-stage to check linting before commiting.

5. Editted README.md.

### CODE REVIEW

File: src/components/ClockFace.jsx [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
-const ClockFace = () => {
+const ClockFace = (props) => {
   const [sessions, setSessions] = useState([1500, 300, 1500, 300]);
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [isActive, setIsActive] = useState(false);
   const [currentSessionIndex, setCurrentSessionIndex] = useState(0);
   const [currentSeconds, setCurrentSeconds] = useState(sessions[0]);
 
+  const [totalSeconds, setTotalSeconds] = useState(currentSeconds);
+  const timeProgress = (currentSeconds / totalSeconds) * 100;
```
Comment (qho-queenie) [inline src/components/ClockFace.jsx:17]: TotalSeconds is the total number of seconds for each session. I use this to calculate the % for the progress bar

File: src/components/ClockFace.jsx [hunk 2]
```diff
@@ -29,23 +35,27 @@ const ClockFace = () => {
...
+      clearInterval(timeInterval);
     }
-    return () => clearInterval(timeInterval)
+    return () => clearInterval(timeInterval);
   }, [isActive, currentSeconds, currentSessionIndex, sessions]);
 
   useEffect(() => {
-    setCurrentSeconds(sessions[currentSessionIndex])
+    setCurrentSeconds(sessions[currentSessionIndex]);
   }, [sessions, currentSessionIndex]);
```
Comment (qho-queenie) [inline src/components/ClockFace.jsx:50]: This is the only new useEffect in this PR

File: src/components/ClockFace.jsx [hunk 3]
```diff
@@ -29,23 +35,27 @@ const ClockFace = () => {
...
 
   useEffect(() => {
-    setCurrentSessionIndex(0)
-    setCurrentSeconds(sessions[0])
+    setTotalSeconds(sessions[currentSessionIndex]);
+  }, [currentSessionIndex, sessions, currentSeconds]);
```
Comment (qho-queenie) [inline src/components/ClockFace.jsx:53]: setting totalSeconds for %

File: src/components/SettingsModal.jsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    </div>
+  );
+};
+
+ModalContext.propTypes = {
+  isModalOpen: PropTypes.boolean,
+  setIsModalOpen: PropTypes.func.isRequired,
+  sessions: PropTypes.arrayOf(PropTypes.number).isRequired,
+  setSessions: PropTypes.func.isRequired,
+};
```
Comment (qho-queenie) [inline src/components/SettingsModal.jsx:109]: I changed the file ext from .js to .jsx, but mostly I just added proptype checks in this file

File: src/components/ClockFace.jsx [hunk 5]
```diff
@@ -29,23 +35,27 @@ const ClockFace = () => {
...
           }
-          setCurrentSeconds(sessions[currentSessionIndex])
+          setCurrentSeconds(sessions[currentSessionIndex]);
           clearInterval(timeInterval);
         }
-      }, 1000)
+      }, 1000);
     } else if (!isActive) {
-      clearInterval(timeInterval)
+      clearInterval(timeInterval);
```
Comment (qho-queenie) [inline src/components/ClockFace.jsx:42]: I know you mentioned a new line for else if and else, but it comes from one of the linting extensions. I can remove the airbnb linting like you said for the next project if it is an overkill, and new line is a better practice.

File: src/components/CircularProgressBar.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './CircularProgressBar.scss';
+
+export const CircularProgressBar = (dialProps) => {
```
Comment (kevinhongw) [inline src/components/CircularProgressBar.jsx:5]: Awesome job here! Very clean component.

File: src/components/ClockFace.jsx (hunk 5 above)
Comment (kevinhongw) [reply src/components/ClockFace.jsx:42]: oh actually I prefer what you have right now. I don't prefer a one line `if` statement tho.
```js
if (x) return 'I do not like this'


if (x) {
  return 'I like this'
}
```

File: src/components/ClockFace.jsx [hunk 7]
```diff
@@ -54,44 +62,63 @@ const ClockFace = () => {
     }
   }, [isModalOpen]);
 
+  const workOrBreak = (currentSessionIndex % 2) === 0 ? 'Pomodoro' : 'Break';
```
Comment (kevinhongw) [inline src/components/ClockFace.jsx]: The name is confusing. You should include the word `label` in it.

File: src/components/SettingsModal.jsx [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    } else if (e.target.name === 'shortSession') {
+      const newSettings = [...localSessions];
+      newSettings[1] = e.target.valueAsNumber * 60;
+      newSettings[3] = e.target.valueAsNumber * 60;
+      setLocalSessions(newSettings);
+    }
+    return null;
+  };
+
+  const formatToMinutes = (seconds) => Math.floor(seconds / 60);
```
Comment (kevinhongw) [inline src/components/SettingsModal.jsx]: This function can be moved out of component.

**Review by kevinhongw — APPROVED** [review]
So sorry about the late review.
This is actually a very high quality PR! Good job!
This project seems quite mature now IMO.

One thing we can try to do now is to schedule a meeting and recreate the project but in a limited time (i.e. 30 mins, 1 hr) as if you're in a interview.

**Pull request merged**